specs/phase1/beacon-chain.md

```diff
@@ -752,298 +374,104 @@ def process_operations(state: BeaconState, body: BeaconBlockBody) -> None:
-                    pending_attestation.crosslink_success = True
-```
-
-###### `verify_empty_shard_transition`
-
-```python
-def verify_empty_shard_transition(state: BeaconState, shard_transitions: Sequence[ShardTransition]) -> bool:
-    """
-    Verify that a `shard_transition` in a block is empty if an attestation was not processed for it.
-    """
-    for shard in range(get_active_shard_count(state)):
-        if state.shard_states[shard].slot != compute_previous_slot(state.slot):
-            if shard_transitions[shard] != ShardTransition():
-                return False
-    return True
-```
-
-###### `process_shard_transitions`
-
-```python
-def process_shard_transitions(state: BeaconState,
-                              shard_transitions: Sequence[ShardTransition],
-                              attestations: Sequence[Attestation]) -> None:
-    # NOTE: We currently set `PHASE_1_FORK_SLOT` to `GENESIS_SLOT` for test vectors.
-    if compute_previous_slot(state.slot) > GENESIS_SLOT:
-        # Process crosslinks
-        process_crosslinks(state, shard_transitions, attestations)
-
-    # Verify the empty proposal shard states
-    assert verify_empty_shard_transition(state, shard_transitions)
-```
-
-##### New default validator for deposits
-
-```python
-def get_validator_from_deposit(state: BeaconState, deposit: Deposit) -> Validator:
-    amount = deposit.data.amount
-    effective_balance = min(amount - amount % EFFECTIVE_BALANCE_INCREMENT, MAX_EFFECTIVE_BALANCE)
-    next_custody_secret_to_reveal = get_custody_period_for_validator(
-        ValidatorIndex(len(state.validators)),
-        get_current_epoch(state),
+    pending_header.votes = bitwise_or(
+        pending_header.votes,
+        attestation.aggregation_bits
     )
 
-    return Validator(
-        pubkey=deposit.data.pubkey,
-        withdrawal_credentials=deposit.data.withdrawal_credentials,
-        activation_eligibility_epoch=FAR_FUTURE_EPOCH,
-        activation_epoch=FAR_FUTURE_EPOCH,
-        exit_epoch=FAR_FUTURE_EPOCH,
-        withdrawable_epoch=FAR_FUTURE_EPOCH,
-        effective_balance=effective_balance,
-        next_custody_secret_to_reveal=next_custody_secret_to_reveal,
-        all_custody_secrets_revealed_epoch=FAR_FUTURE_EPOCH,
+    # Check if the PendingShardHeader is eligible for expedited confirmation
+    # Requirement 1: nothing else confirmed
+    all_candidates = [
+        c for c in pending_headers if
+        (c.slot, c.shard) == (pending_header.slot, pending_header.shard)
+    ]
+    if True not in [c.confirmed for c in all_candidates]:
+        # Requirement 2: >= 2/3 of balance attesting
+        participants = get_attesting_indices(state, data, pending_commitment.votes)
+        participants_balance = get_total_balance(state, participants)
+        full_committee = get_beacon_committee(state, data.slot, data.shard)
+        full_committee_balance = get_total_balance(state, full_committee)
```

Comment:
Should this be all, or only online?

Reply:
I'm inclined to say all. We already have a second case to cover the ">1/3 offline" case which is to wait until the end of the next epoch and take the candidate with the most votes (which could be empty).